app/Libraries/Markdown/Indexing/Extension.php

```diff
@@ -9,6 +9,7 @@
 use League\CommonMark\Block\Element as Block;
 use League\CommonMark\ConfigurableEnvironmentInterface;
 use League\CommonMark\Extension\ExtensionInterface;
+use League\CommonMark\Extension\Footnote\Node as FootnoteExtension;
```

Comment:
seeing how many renderers are being overridden here (are they even necessary?) please look into updating commonmark to latest version (2.0) first.

Reply:
- `FootnoteContainerRenderer` is used to match `osu-md__list` structure.
- `FootnoteListItemRenderer` is used to match `osu-md__list-item` structure.
- `FootnoteRefRenderer` and `FootnoteRefRenderer` is used to tweak the content a bit.